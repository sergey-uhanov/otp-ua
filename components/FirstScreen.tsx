import React from 'react';
import style from '@/styles/page/HomePage.module.css';
import Image from 'next/image';
import banner from '@/public/img/bPOGNwCWuXYCecK7QrcHM.jpg';

import LinkBorderButton from './UI/buttons/LinkBorderButton';
import FillLinkButton from './UI/buttons/FillLinkButton';
interface FirstScreenProps {}
const FirstScreen = ({}: FirstScreenProps) => {
  return (
    <section className={style.firstScreen}>
      <div className={style.banner}>
        <div style={{ display: 'flex', flexDirection: 'column' }}>
          <Image
            alt="Mountains"
            src={banner}
            sizes="100vw"
            style={{
              width: '100%',
              height: '630px',
              minWidth: '300px',
              maxWidth: '360px',
              objectFit: 'cover',
            }}
          />
        </div>
      </div>
      <div className={style.firstScreen_text_bloc}>
        <h1 className={style.firstScreen_title}>
          Створіть свої ідеальні опитування та тести
        </h1>
        <p className={style.firstScreen_content}>
          Наш інтуїтивний конструктор опитань та тестів допоможе вам створювати
          ефективні інструменти для збору цінної інформації та взаємодії з вашою
          аудиторією.
        </p>
        <div className={style.firstScreen_btn_container}>
          <FillLinkButton url="/create-page">
            Створити опитування
          </FillLinkButton>
          <LinkBorderButton url="/">Преглянути приклади</LinkBorderButton>
        </div>
      </div>
    </section>
  );
};

export default FirstScreen;
