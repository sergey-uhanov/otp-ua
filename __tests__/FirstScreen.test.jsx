import { render, screen } from '@testing-library/react';
import FirstScreen from '@/components/FirstScreen';
import '@testing-library/jest-dom';

describe('FirstScreen', () => {
  it('renders the component with heading, paragraph and buttons', () => {
    render(<FirstScreen />);

    // Проверка заголовка
    const heading = screen.getByRole('heading', {
      name: /створіть свої ідеальні опитування та тести/i,
    });
    expect(heading).toBeInTheDocument();

    // Проверка текста параграфа
    const paragraph = screen.getByText(
      /наш інтуїтивний конструктор опитань та тестів/i
    );
    expect(paragraph).toBeInTheDocument();

    // Проверка кнопки "Створити опитування"
    const createButton = screen.getByRole('link', {
      name: /створити опитування/i,
    });
    expect(createButton).toBeInTheDocument();
    expect(createButton).toHaveAttribute('href', '/create-page');

    // Проверка кнопки "Преглянути приклади"
    const viewExamplesButton = screen.getByRole('link', {
      name: /преглянути приклади/i,
    });
    expect(viewExamplesButton).toBeInTheDocument();
    expect(viewExamplesButton).toHaveAttribute('href', '/');
  });

  it('renders the banner image with alt text', () => {
    render(<FirstScreen />);

    // Проверка изображения
    const image = screen.getByAltText('Mountains');
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute(
      'src',
      expect.stringContaining('/_next/image?url=%2Fimg.jpg&w=3840&q=75')
    );
  });
});
describe('FirstScreen Snapshot Test', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<FirstScreen />);
    expect(asFragment()).toMatchSnapshot();
  });
});
