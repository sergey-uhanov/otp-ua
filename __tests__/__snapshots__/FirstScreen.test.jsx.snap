// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstScreen Snapshot Test matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="firstScreen"
  >
    <div
      class="banner"
    >
      <div
        style="display: flex; flex-direction: column;"
      >
        <img
          alt="Mountains"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="color: transparent; width: 100%; height: 630px; min-width: 300px; max-width: 360px; object-fit: cover;"
          width="40"
        />
      </div>
    </div>
    <div
      class="firstScreen_text_bloc"
    >
      <h1
        class="firstScreen_title"
      >
        Створіть свої ідеальні опитування та тести
      </h1>
      <p
        class="firstScreen_content"
      >
        Наш інтуїтивний конструктор опитань та тестів допоможе вам створювати ефективні інструменти для збору цінної інформації та взаємодії з вашою аудиторією.
      </p>
      <div
        class="firstScreen_btn_container"
      >
        <a
          class="button"
          href="/create-page"
        >
          Створити опитування
        </a>
        <a
          class="button"
          href="/"
        >
          Преглянути приклади
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
